internal/attestations/attestations.go

```diff
@@ -36,6 +43,24 @@ type Attestations struct {
 	// `<ref-path>/<commit-id>`, where `ref-path` is the absolute ref path, and
 	// `commit-id` is the ID of the merged commit.
 	githubPullRequestAttestations map[string]gitinterface.Hash
+
+	// githubPullRequestApprovalAttestations stores the blob ID of a GitHub pull
+	// request approval attestation for the change it applies to. The key is a
+	// path of the form `<ref-path>/<from-id>-<to-id>`, where `ref-path` is the
+	// absolute ref path such as `refs/heads/main` and `from-id` and `to-id`
+	// determine how the ref in question moved. For example, the key
+	// `refs/heads/main/<commit-A>-<tree-B>` indicates the pull request updated
+	// `refs/heads/main` from `commit-A` to a commit with `tree-B`.
+	githubPullRequestApprovalAttestations map[string]gitinterface.Hash
+
+	// githubPullRequestApprovalIndex is stored in-memory. It maps a GitHub pull
+	// request review ID to the gittuf identifier for a review,
+	// `<ref-path>/<from-id>-<to-id>`. We need this because when a review is
+	// dismissed, we need to unambiguously know what the review applied to when
+	// it was first submitted, which we cannot do with the information at the
+	// time of dismissal. This is serialized to the attestations namespace as a
+	// special blob in the githubPullRequestApprovalAttestations tree.
+	githubPullRequestApprovalIndex map[int64]string
```

Comment:
I'm a bit skeptical of this - I think we need to include remote information of some kind here if we're going to be keying off the pull request id, otherwise it's ambiguous across repos.

I'm also skeptical of using int64 if we ever want to expand this out to other types of presubmit attestations.

Reply:
just emphasizing, this isn't the PR number but rather the PR ID. I thought that was unique across repositories, but I can fix this to have more data points. TBF, there's also no easy way to look up a PR strictly by ID I think, so having additional information would be good.

int64 is because I'm using the ID field exactly as returned by go-github, so it matches what's used for the field in the API response. Do we expect presubmit attestations to be concentrated in how they're stored in refs/gittuf/attestations? Do you mean if we're supporting gitlab in addition to github?

Reply:
Generalized this layer to support multiple code review systems.